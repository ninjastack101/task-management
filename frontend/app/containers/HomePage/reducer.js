import produce from 'immer';
import { LOAD_TASKS } from './constants';

// The initial state of the App
export const initialState = {
  tasks: [],
};

/* eslint-disable default-case, no-param-reassign */
const homeReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case LOAD_TASKS:
        draft.tasks = action.tasks;
        break;
    }
  });

export default homeReducer;
