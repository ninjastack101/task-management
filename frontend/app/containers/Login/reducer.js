import produce from "immer";
import { SIGNUP_SUCCESS, LOGOUT } from "./constants";

// The initial state of the App
export const initialState = {
  username: "",
  userId: ""
};

/* eslint-disable default-case, no-param-reassign */
const loginReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case SIGNUP_SUCCESS:
        draft.userId = action.id;
        draft.username = action.username;
        break;
      case LOGOUT:
        draft.userId = "";
        draft.username = "";
        break;
    }
  });

export default loginReducer;
