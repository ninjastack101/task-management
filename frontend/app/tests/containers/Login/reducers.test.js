import reducer from '../../../containers/Login/reducer';
import {
  SIGNUP_SUCCESS,
  LOGOUT
}
  from '../../../containers/Login/constants';

describe('Sign up user reducer', () => {

  let initialState = {
    username: "",
    userId: ""
  };

  it('Should return the initial state', (done) => {
    expect(reducer(undefined, {})).toEqual(initialState);
    done();
  });

  it('should handle SIGNUP_SUCCESS', (done) => {

    const startAction = {
      type: SIGNUP_SUCCESS,
      userId: "somestring",
      username: "abc",
    };
    expect(reducer(initialState, startAction)).toMatchSnapshot();
    done();
  });

  it('should handle LOGOUT', (done) => {

    const startAction = {
      type: LOGOUT,
      userId: "",
      username: ""
    };
    expect(reducer(initialState, startAction)).toMatchSnapshot();
    done();
  });
});
