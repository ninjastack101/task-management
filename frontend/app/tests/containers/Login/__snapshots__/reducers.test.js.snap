// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up user reducer should handle LOGOUT 1`] = `
Object {
  "userId": "",
  "username": "",
}
`;

exports[`Sign up user reducer should handle SIGNUP_SUCCESS 1`] = `
Object {
  "userId": undefined,
  "username": "abc",
}
`;
